memtable/write_buffer_manager.cc

```diff
@@ -115,52 +147,26 @@ void WriteBufferManager::FreeMemWithCache(size_t mem) {
   s.PermitUncheckedError();
 }
 
-void WriteBufferManager::BeginWriteStall(StallInterface* wbm_stall) {
+bool WriteBufferManager::MaybeBeginWriteStall(StallInterface* wbm_stall) {
   assert(wbm_stall != nullptr);
-  assert(allow_stall_);
-
-  // Allocate outside of the lock.
```

Comment:
Question 3: just to clarify, this "allocate" means allocate of the list of pointers `new_node` not the actual StallInterface object wbm_stall right? Same for all the de-allocation done outside the lock.